src/credentials/providers/sharedCredentialsProvider.ts

```diff
@@ -57,6 +58,15 @@ export class SharedCredentialsProvider implements CredentialsProvider {
         return this.profile[SHARED_CREDENTIAL_PROPERTIES.REGION]
     }
 
+    /**
+     * Decides if the credential is the kind that may be auto-connected at
+     * first use (in particular, credentials that may prompt, such as SSO/MFA,
+     * should _not_ attempt to auto-connect).
+     */
+    public canAutoConnect(): boolean {
```

Comment:
Please add a test. If/when this method's logic grows out, its easier to keep on top of when the tests are already written.